src/core/appdir.cpp

```diff
@@ -834,7 +834,15 @@ namespace linuxdeploy {
                     if (!d->deployElfDependencies(sharedLibrary))
                         return false;
 
-                    d->setElfRPathOperations[sharedLibrary] = "$ORIGIN";
+                    const auto rpath = elf_file::ElfFile(sharedLibrary).getRPath();
+                    auto rpathList = util::split(rpath, ':');
+                    if (std::find(rpathList.begin(), rpathList.end(), "$ORIGIN") == rpathList.end())
+                    {
+                        rpathList.push_back("$ORIGIN");
+                        d->setElfRPathOperations[sharedLibrary] = util::join(rpathList, ":");
+                    } else {
+                        d->setElfRPathOperations[sharedLibrary] = rpath;
```

Comment:
Just an idea (I don't insist on it), couldn't we spare this one and just call `d->setElfRPathOperations[sharedLibrary] = util::join(rpathList, ":");` once after the `if`?

Reply:
We can, but it's call `join(rpathList)` even than rpath not changed.

Reply:
It won't make a difference, I guess, just saves two lines. Let's keep it like this. Thanks!